scripts/lint_prowjobs.go

```diff
@@ -0,0 +1,279 @@
+/*
+Copyright 2020 Amazon.com Inc. or its affiliates. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+
+	prowapi "k8s.io/test-infra/prow/config"
+	yaml "sigs.k8s.io/yaml"
+)
+
+type PresubmitConstants struct {
+	Bucket                   string
+	Cluster                  string
+	ServiceAccountName       string
+	DefaultMakeTarget        string
+	CniMakeTarget            string
+	HelmMakeTarget           string
+	ReleaseToolingMakeTarget string
+}
+
+type PostsubmitConstants struct {
+	Bucket                       string
+	Cluster                      string
+	DefaultServiceAccountName    string
+	AthensHelmServiceAccountName string
+	DocsServiceAccountName       string
+	MakeTarget                   string
+}
+
+type Presubmits struct {
+	PresubmitJobs map[string][]prowapi.Presubmit `json:"presubmits,omitempty"`
+}
+
+type Postsubmits struct {
+	PostsubmitJobs map[string][]prowapi.Postsubmit `json:"postsubmits,omitempty"`
+}
+
+var presubmitErrors map[string][]string
+var postsubmitErrors map[string][]string
+var presubmitErrorCount, postsubmitErrorCount = 0, 0
```

Comment:
This would be more intuitive if you didn't use globals here, the methods could return the map. Also, these counts are probably not necessary, you could just len(postsubmitErrors)

Reply:
`postsubmitErrors` stores the errors against each file, so the map itself will never be empty since we'll have the files in the PR as keys. I could use a flag to check if any file has a non-zero error count.